packages/jest-leak-detector/src/index.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+import v8 from 'v8';
+import vm from 'vm';
+import prettyFormat from 'pretty-format';
+
+export default class {
+  _isReferenceBeingHeld: boolean;
+
+  constructor(value: any) {
+    if (this._isPrimitive(value)) {
+      throw new TypeError(
+        [
+          'Primitives cannot leak memory.',
+          'You passed a ' + typeof value + ': <' + prettyFormat(value) + '>',
+        ].join(' '),
+      );
+    }
+
+    let weak;
+
+    try {
+      // eslint-disable-next-line import/no-extraneous-dependencies
+      weak = require('weak');
+    } catch (err) {
+      if (!err || err.code !== 'MODULE_NOT_FOUND') {
+        throw err;
+      }
+
+      throw new Error(
+        'The leaking detection mechanism requires the "weak" package to be installed and work. ' +
+          'Please install it as a dependency on your main project',
+      );
+    }
+
+    weak(value, () => (this._isReferenceBeingHeld = false));
+    this._isReferenceBeingHeld = true;
+
+    // Ensure value is not leaked by the closure created by the "weak" callback.
+    value = null;
+  }
+
+  isLeaking(): boolean {
+    this._runGarbageCollector();
+
+    return this._isReferenceBeingHeld;
+  }
+
+  _runGarbageCollector() {
```

Comment:
you can add `private` to the functions with `_` in their name

Reply:
👍